spec/PasswordPolicy.spec.js

```diff
@@ -757,4 +759,233 @@ describe("Password Policy: ", () => {
+          done();
+        });
+      }).catch((error) => {
+        jfail(error);
+        fail('Signup failed.');
+        done();
+      });
+    })
+  });
+
+  it('should fail if logged in after password expires', (done) => {
+    const user = new Parse.User();
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      passwordPolicy: {
+        daysBeforeExpiry: 0.000005787 // 0.5 sec
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      user.setUsername("user1");
+      user.setPassword("user1");
+      user.set('email', 'user1@parse.com');
+      user.signUp().then(() => {
+        // wait for a bit more than the validity duration set
+        setTimeout(() => {
+          Parse.User.logIn("user1", "user1").then(() => {
+            fail("logIn should have failed");
+            done();
+          }).catch((error) => {
+            expect(error.code).toEqual(Parse.Error.OBJECT_NOT_FOUND);
+            expect(error.message).toEqual('Your password has expired. Please reset your password.');
+            done();
+          });
+        }, 1000);
+      }).catch((error) => {
+        jfail(error);
+        fail('Signup failed.');
+        done();
+      });
+    });
+  });
+
+  it('should apply password expiry policy to existing user upon first login after policy is enabled', (done) => {
+    const user = new Parse.User();
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      user.setUsername("user1");
+      user.setPassword("user1");
+      user.set('email', 'user1@parse.com');
+      user.signUp().then(() => {
+        Parse.User.logOut();
+        reconfigureServer({
+          appName: 'passwordPolicy',
+          passwordPolicy: {
+            daysBeforeExpiry: 0.000005787 // 0.5 sec
+          },
+          publicServerURL: "http://localhost:8378/1"
+        }).then(() => {
+          Parse.User.logIn("user1", "user1").then((u) => {
+            Parse.User.logOut();
+            // wait for a bit more than the validity duration set
+            setTimeout(() => {
+              Parse.User.logIn("user1", "user1").then(() => {
+                fail("logIn should have failed");
+                done();
+              }).catch((error) => {
+                expect(error.code).toEqual(Parse.Error.OBJECT_NOT_FOUND);
+                expect(error.message).toEqual('Your password has expired. Please reset your password.');
+                done();
+              });
+            }, 2000);
+          }).catch((error) => {
+            jfail(error);
+            fail('Login failed.');
+            done();
+          });
+        });
+      }).catch((error) => {
+        jfail(error);
+        fail('Signup failed.');
+        done();
+      });
+    });
+
+  });
+
+  it('should reset password timestamp when password is reset', done => {
+    var user = new Parse.User();
```

Comment:
@bhaskaryasa 
Can you pls use `let` instead of `var` for better scoping of variables?